University-advisor/Forms/MainForm.cs

```diff
@@ -226,11 +227,25 @@ private void ProgrammesGrid_CellClick(object sender, DataGridViewCellEventArgs e
                 return;
             int selectedId = (int)dgv.CurrentRow.Cells["Id"].Value;
             selectedCourse = selectedId;
+            string userId = GetUserId(currentUser);
+            var reviews = SqlDriver.Fetch($"SELECT * FROM coursereviews WHERE userId = {userId} AND courseId = {selectedCourse} LIMIT 1");
+            if (reviews != null && reviews.Count > 0)
+            {
+                MessageBox.Show(Messages.reviewAlreadySubmitted);
+                return;
+            }
```

Comment:
In order to avoid code duplication maybe this part could be separated into a method and then called from specific places in the code only changing the SQL query?

Reply:
Great idea. Implemented it.